tests/components/metoffice/test_weather.py

```diff
@@ -35,13 +37,74 @@ async def test_site_cannot_connect(hass, requests_mock, legacy_patchable_time):
     await hass.config_entries.async_setup(entry.entry_id)
     await hass.async_block_till_done()
 
-    assert hass.states.get("weather.met_office_wavertree") is None
+    assert hass.states.get("weather.met_office_wavertree_3hourly") is None
+    assert hass.states.get("weather.met_office_wavertree_daily") is None
     for sensor_id in WAVERTREE_SENSOR_RESULTS:
         sensor_name, sensor_value = WAVERTREE_SENSOR_RESULTS[sensor_id]
         sensor = hass.states.get(f"sensor.wavertree_{sensor_name}")
         assert sensor is None
 
 
+@patch(
+    "datapoint.Forecast.datetime.datetime",
+    NewDateTime,
+)
+async def test_site_list_cannot_update(hass, requests_mock, legacy_patchable_time):
+    """Test we handle cannot connect error."""
+
+    # all metoffice test data encapsulated in here
+    mock_json = json.loads(load_fixture("metoffice.json"))
+    all_sites = json.dumps(mock_json["all_sites"])
+    wavertree_hourly = json.dumps(mock_json["wavertree_hourly"])
+    wavertree_daily = json.dumps(mock_json["wavertree_daily"])
+
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text=all_sites)
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=3hourly", text=wavertree_hourly
+    )
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=daily", text=wavertree_daily
+    )
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=METOFFICE_CONFIG_WAVERTREE,
+    )
+    entry.add_to_hass(hass)
+    await hass.config_entries.async_setup(entry.entry_id)
+    await hass.async_block_till_done()
+
+    entity = hass.states.get("weather.met_office_wavertree_3_hourly")
+    assert entity
+
+    entity = hass.states.get("weather.met_office_wavertree_daily")
+    assert entity
+
+    requests_mock.get("/public/data/val/wxfcs/all/json/sitelist/", text="")
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=3hourly", text=wavertree_hourly
+    )
+    requests_mock.get(
+        "/public/data/val/wxfcs/all/json/354107?res=daily", text=wavertree_daily
+    )
+
+    # Make site list unavailable
+    # Datapoint API caches requests for 1 hour so we need to reset the timer here
+    data = hass.data[DOMAIN][entry.entry_id][METOFFICE_DATA]
+    data._datapoint.forecast_sites_last_update = 0
+    await data.async_update_site()
```

Comment:
This call only seems to happen during integration set up. Why call it explicitly here?

Reply:
Pretty much the only reason this test even exists is to add test coverage to the case where site doesn't exist when `_update_hourly` or `_update_daily` is called but site doesn't exist. This is a very hypothetical situation and the only way I could reproduce it is by explicitly calling `async_update_site` again.